src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestCommentTrigger.java

```diff
@@ -56,29 +58,33 @@ public void isMatch() {
                         if (gitLabSCMSource.getProjectId() == getPayload().getMergeRequest().getTargetProjectId()
                                 && isTrustedMember(gitLabSCMSource, sourceContext.getOnlyTrustedMembersCanTrigger())) {
                             for (Job<?, ?> job : owner.getAllJobs()) {
-                                if (mergeRequestJobNamePattern.matcher(job.getName()).matches()) {
-                                    String expectedCommentBody = sourceContext.getCommentBody();
-                                    Pattern pattern = Pattern.compile(expectedCommentBody,
-                                            Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
-                                    if (commentBody == null || pattern.matcher(commentBody)
-                                            .matches()) {
-                                        ParameterizedJobMixIn.scheduleBuild2(job, 0,
-                                                new CauseAction(
-                                                        new GitLabMergeRequestCommentCause(commentUrl, getPayload())));
-                                        LOGGER.log(Level.INFO,
-                                                "Triggered build for {0} due to MR comment on {1}",
-                                                new Object[] {
-                                                        job.getFullName(),
-                                                        getPayload().getProject().getPathWithNamespace()
-                                                });
-                                    } else {
-                                        LOGGER.log(Level.INFO,
-                                                "MR comment does not match the trigger build string ({0}) for {1}",
-                                                new Object[] { expectedCommentBody, job.getFullName() });
+                                if (MultiBranchProject.class.isAssignableFrom(job.getParent().getClass()) ) {
+                                    MultiBranchProject parentJob = (MultiBranchProject) job.getParent();
+                                    if (parentJob.getSCMSource(gitLabSCMSource.getId()) == gitLabSCMSource
```

Comment:
I try to check the link between the source of the payload and the job in the for.

I don't know if there is a better solution